feat: FAQ - Add Network and Nodes Chapter

fhildeb opened the pull request:
Removes the old Network and Validator pages and sets up the new Network and Nodes folder, including the following pages:

- Blockchain Architecture
- Network Configuration
- Peer Connections
- Node Setup
- Validators
- Security
- Staking

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+# Blockchain Architecture
+
+### What is LUKSOs Network Architecture?
```
why LUKSOs? no need for the final 's' i think

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+LUKSO's network architecture is based on the **Proof of Stake** consensus protocol, which relies on validators to secure the network and maintain consensus. The protocol is combined with two fundamental necessities to allow a secure and efficient network.
+
+The first necessity is **achieving the finality** of changes made to the network. This ensures that once a block is added to the blockchain, its information is immutable and cannot be changed. Finality is managed by the Casper the Friendly Finality Gadget (**Casper FFG**), a consensus mechanism designed to achieve this finality.
```
"immutable and cannot be changed" -> it's the same thing
"Finality is managed by the Casper the Friendly Finality Gadget (**Casper FFG**)" -> remove 'the' just before Casper

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+### What is Proof of Stake?
+
+PoS is a consensus mechanism many blockchain networks use to **validate and add new transactions**. Unlike Proof of Work, which requires miners to solve complex mathematical problems to add new blocks to the blockchain, PoS relies on the amount of cryptocurrency a person holds and is willing to stake as collateral.
+
+In a PoS blockchain, validators are chosen to create a new block based on their **stake in the network**. The more cryptocurrency a person holds, the chances of being selected as a validator increase. Once chosen, validators validate transactions and add them to the blockchain. For their service, validators are rewarded with transaction fees and potentially new coins.
```
replace "cryptocurrency" by "native tokens"

kalote replied:
also, this "The more cryptocurrency a person holds, the chances of being selected as a validator increase" i'm not sure:
- if you mean that with more native token you can create more validators, thus having more chances to validate blocks, ok, but it needs to be rephrased
- if you mean having more than 32LYX will give you more chances to be selected, then it's not true

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+One of the main advantages of PoS over PoW is its **energy efficiency**. While PoW requires massive computational resources and energy consumption, PoS achieves consensus with minimal energy use. The efficiency makes PoS more sustainable and environmentally friendly for blockchain networks.
+
+Another advantage of PoS is the security it provides. Since **validators have a significant investment** in the networks coin, they are incentivized to act honestly. If a validator tries to manipulate the system or validate fraudulent transactions, they risk losing their stake, making attacks on the network costly and, therefore, less likely.
```
replace "coins" with something else (token / native token can be good)

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+### What EVM Protocol Upgrades are live on LUKSO?
+
+The LUKSO Mainnet is up to date with [every Ethereum fork](https://ethereum.org/en/history/) until Shanghai, which was released in 2023. Except for Shapella (Capella and Shanghai), all upgrades were integrated within the mainnet launch.
```
this paragraph is not up to date I believe, as we did the shapella fork few days ago.

Hugoo replied:
I think Johann is right, the paragraph says Shapella is not activated, but it was a few days ago:
https://github.com/lukso-network/network-configs/blob/main/mainnet/shared/genesis.json#L18

<img width="725" alt="image" src="https://github.com/lukso-network/docs/assets/477945/a44bd6e5-9dfd-4224-b640-4125f0783406">

Also @fhildeb for things like that, let's let the person who made the comment write the conversation as resolved.

fhildeb replied:
@Hugoo Alright. But the sentence says that all forks are activated... as Shanghai is the last fork included in Shapella. Where does it state Shapella is not activated? The answer states that Shapella was not activated from genesis on, but later. Maybe I should add the timestamp to clear things up.

Hugoo replied:
Ok i see, i think if both Johann and I got confused, maybe let's just make it more obvious.

Shapella was activated on XXX.

I read it fast and i read: "All were activated EXCEPT shapella".

Let's do maybe:

```suggestion
The LUKSO Mainnet is up to date with [every Ethereum fork](https://ethereum.org/en/history/) until Shanghai, which was released in 2023. All upgrades were integrated within the mainnet launch, except for Shapella (Capella and Shanghai) which was activated on the 11th of July.
```

fhildeb replied:
Great. I just rephrased the sentence a bit more to also include the correct block times. (so there is no confusion when we add future forks)

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+### Why is client diversity so important?
+
+Having a variety of clients in a blockchain network and improving our client diversity is critically important. Client diversity refers to utilizing different software clients in a blockchain network developed by various teams and in other programming languages to improve the following things:
```
"to improve the following" (remove things)

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+- **Security and Resilience**: Client diversity increases the robustness of the network. If there's a bug in one client, it doesn't bring down the entire network because other clients can continue to operate. This decentralization and redundancy is a fundamental aspect of blockchain security and resilience.
+- **Decentralization and Governanc**e: Client diversity promotes decentralization in the development and governance of the Ethereum network. It prevents any team or entity from having too much influence over the network's growth.
+
+For Ethereum, **client diversity has proven essential** in maintaining the network's robustness during several incidents. One of the most notable incidents were the Shanghai DoS Attacks in 2016, the OpenEthereum Consensus Bug in 2020, and the Prysm Client Incident in 2023. You can read more about them in the [extended client section](https://docs.luksoverse.io/docs/mainnet/complete-node-guide/blo ...
```
links to those notable incidents? if the link to luksoverse is supposed to cover those incidents, use the anchor to ethereum-history -> https://docs.luksoverse.io/docs/mainnet/complete-node-guide/blockchain-clients/client-setups/#ethereums-history
I would prefer a link to something not LUKSO related when we talk about previous incident 😓

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+### How does Shapella enable validator withdrawals?
+
+Shanghai and Capella (Shapella) are the names of an Ethereum hard fork. Shanghai is the fork's name on the execution client side, and Capella is the upgrade name on the consensus layer client side. The included EIPs can be found [here](https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/shanghai.md). The significant new features enable the **withdrawal of the ...
```
"Shanghai and Capella (Shapella) are the names of an Ethereum hard fork" => "... is the name of ..."

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+### What is the London Update?
+
+Each operation in the EVM requires a certain amount of gas, which is paid in the blockchain's native coin. The cost of gas is a **crucial part of the network's incentive structure**, deterring spam and incentivizing miners to confirm transactions.
```
"native coin" -> native token

fhildeb replied:
see comment/source above, i will keep coin

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+### What is slashing?
+
+In the Proof of Stake consensus context, the slasher functionality **discourages validators from behaving dishonestly or maliciously**. If a validator behaves in a way that could compromise the network's integrity—like trying to manipulate the transaction history or proposing conflicting blocks—they can be slashed. When a validator gets slashed, a portion of its staked LYX or LYXt is removed, e,g ...
```
- add space before / after the dashes
- "Additionally, they are **ejected from the validator set**, losing their ability to participate in the consensus process and earn further rewards." -> we should add that there is a condition related to ejection (less than 16 LYX)

fhildeb replied:
we actually have a question for that in an other file, let me link them

kalote commented on docs/faq/network/blockchain-architecture.md:
```diff
@@ -0,0 +1,103 @@
+
+### What are the expected block and epoch times?
+
+The LUKSO network will run with an expected block time of 12 seconds with 12 seconds per slot. An epoch contains 32 slots.
```
"with an expected block time of 12 seconds with 12 seconds per slot." => weird phrasing. Maybe we can remove the whole part after the first "12 seconds"?

kalote commented on docs/faq/network/network-configuration.md:
```diff
@@ -0,0 +1,56 @@
+
+The following ports will need to be opened to run the blockchain node correctly:
+
+| CLIENT     | DESCRIPTION                               | PORT  | TCP | UDP |
+| ---------- | ----------------------------------------- | ----- | --- | --- |
+| GETH       | Execution Chain Data Channel              | 30303 | X   |     |
+| GETH       | Execution Chain Discovery                 | 30303 |     | X   |
+| ERIGON     | Execution Chain Data Channel              | 30303 | X   |     |
+| ERIGON     | Execution Chain Discovery                 | 30303 |     | X   |
+| LIGHTHOUSE | Beacon Communication and Data             | 9000  | X   | X   |
+| PRYSM      | Beacon Gossip, Requests, and Responses    | 13000 | X   |     |
+| PRYSM      | Beacon Discovery, Requests, Data Exchange | 12000 |     | X   |
```
I will remove the client + description. Just answer the question:
- which ports must be open?

30303 TCP + UDP
12000 UDP
13000 TCP
(Lighthouse only) 9000 TCP + UDP

fhildeb replied:
if we remove the description and clients, there will be more questions about why they need to be open and which clients they are used for. we need to create these answers so that there are no further questions, freeing our resources.

IMO we should keep them

kalote commented on docs/faq/network/network-configuration.md:
```diff
@@ -0,0 +1,56 @@
+
+### Do I need a stable internet connection to run a validator?
+
+Running an EVM Proof of Stake validator node at home requires a stable and reliable internet connection with sufficient network speeds and low latency. A stable internet connection is crucial to ensure optimal performance and maintain the validator's active status within the network. Here's why speeds and latency are essential:
+
+- **Blockchain Synchronization**: A fast and stable internet connection is required to synchronize the blockchain data with the Ethereum network efficiently. Slow download speeds can lead to longer sync times, which may prevent your validator from actively participating in the consensus process, causing missed opportunities for rewards.
+
+- **Attestations and Proposals**: As a validator, your node creates attestations and proposes blocks. A low-latency connection ensures that your attestations and proposals reach the network promptly, increasing the likelihood of inclusion in the final chain and earning rewards.
+
+- **Missed Events and Penalties**: In Ethereum PoS, validators can be penalized for failing to participate in the consensus process. If your network connection is slow or unstable, it could cause missed attestations or proposals, resulting in penalties and reduced rewards.
+  Network Resilience: A reliable internet connection with low latency helps your validator node stay connected and recover quickly from temporary network outages. Strength reduces the risk of being isolated from the rest of the network and missing important consensus events.
```
Can be simplified:

Do I need a stable internet?

Yes

fhildeb replied:
I've added the yes as a short answer beforehand. However, we faced several questions about network connection, latency, up-, and downtime. This extended answer should cover them all and explain why it's essential. Same view as on the question before: We want to answer it in a way where there are no more open questions in that regard.

kalote commented on docs/faq/network/network-configuration.md:
```diff
@@ -0,0 +1,56 @@
+- **Missed Events and Penalties**: In Ethereum PoS, validators can be penalized for failing to participate in the consensus process. If your network connection is slow or unstable, it could cause missed attestations or proposals, resulting in penalties and reduced rewards.
+  Network Resilience: A reliable internet connection with low latency helps your validator node stay connected and recover quickly from temporary network outages. Strength reduces the risk of being isolated from the rest of the network and missing important consensus events.
+
+### What are the minimal network speeds to run a node?
+
+Sufficient network speeds and low latency are essential for running an Ethereum PoS validator node at home. It ensures optimal performance, maximizes reward potential, and minimizes the risk of penalties or missed attestations.
+
+A minimum download speed of `10-15 Mbps` is recommended, with higher rates being more desirable for faster blockchain synchronization and data transfer. Upload speeds should be at least `2-4 Mbps` to ensure your attestations and proposals can be sent to the network quickly. Regarding latency, a connection below `30-100 ms` is ideal for guaranteeing timely participation in the consensus process an ...
+
```
Line 10, you said that speed doesn't matter as long as the validator answer in 12s , but in this paragraph (and also line 39), you state that speed is important.

fhildeb replied:
Every 12 seconds is the new block in which the network has to receive the answer. I'm going to update the description...

> In summary, while the block time of 12 seconds defines the average time between blocks, having good latency remains important for timely synchronization, transaction propagation, and network participation.

kalote commented on docs/faq/network/network-configuration.md:
```diff
@@ -0,0 +1,56 @@
+
+### How much data traffic does my node consume?
+
+Based on [community research](https://docs.luksoverse.io/), the node will have a daily upload and download of around `40-45 GB` each, totaling approximately `80-90 GB` of internet traffic daily. Based on these metrics, your node will have average data traffic of around `600 GB` per week and around `2,7 TB` per month. Archive nodes will get similar numbers but store data more consistently and in m ...
```
"[community research](https://docs.luksoverse.io/)" -> this is not a link to a report of some sort, nor an analysis of the data traffic.
Remove it, or link a real report

fhildeb replied:
I will remove the link, you're right, it's not a report. I think we don't want to add a PDF with router screenshots somewhere and just state that it was researched by multiple validators. I could not find a true "report" for Ethereum clients

kalote commented on docs/faq/network/node-security.md:
```diff
@@ -0,0 +1,40 @@
+
+If someone could log into your node, he can **only access your validator keys if he also knows your validator wallet password**. By gaining access to your wallet password, the attacker could act maliciously by forcing slashable events or restarting the client with a different recipient fee address.
+
+Machine access is risky if you set up automation with a password file or use Lighthouse- as the client stores the wallet password within the config file during the initial startup. Therefore, it's a must-have to restrict access to your node **using key authentication** and **restrict read and write access of files to a specific user**. This way, no one can see or change your configurations withou ...
```
"it's a must-have to restrict access to your node **using key authentication** and **restrict read and write access of files to a specific user**" -> replace with

it's a must-have to restrict access to your node **using key authentication** and **read and write access to files for a specific user**

kalote commented on docs/faq/network/node-setup.md:
```diff
@@ -0,0 +1,106 @@
+
+The [LUKSO Launchpad](https://deposit.mainnet.lukso.network/) will streamline the process of generating deposit data and transferring validator funds. However, the node setup and maintenance will require using the terminal for both [LUKSO CLI](https://github.com/lukso-network/tools-lukso-cli) and [Docker](https://github.com/lukso-network/network-docker-containers).
+
+### How can I check if my node is working fine?
```
using the consensus explorer also to find if your validators are up and running

fhildeb replied:
This is the node setup. But agree, it might be a good addition for the validator section

kalote commented on docs/faq/network/staking.md:
```diff
@@ -0,0 +1,106 @@
+
+The [LUKSO CLI](https://docs.lukso.tech/networks/mainnet/running-a-node) is a great starting point for setting up a validator node for staking. However, people must set up their server system correctly and install custom monitoring. Additionally, technical people could also use Docker or manual setups.
+
+For non-technical people, we suggest waiting until the DappNode integration. If you do not want to run your node, please wait until the first external staking services are deployed on the LUKSO mainnet.
```
link to dappnode

fhildeb commented:
Thanks, Johann, for taking the time 🙏🏻😙 I applied your suggestions and clarified the answers. Also added redirects and modified the headings as in the other sections

Hugoo commented:
sadly we now have a few conflicts bc we merged the other PRs :/

should be easy to fi anyway
